### DISCUSSION TITLE
How to model shared IPMI interface?

### DISCUSSION THREAD

**Discussion by llamafilm** [netbox-community/netbox#17001] [Q&A]
Most Supermicro servers have 3 physical interfaces:
- LAN 1
- LAN 2
- IPMI

In the default configuration (which I typically use), out-of-band IPMI is shared with LAN 1, so the switch sees 2 IPs and MACs on the same interface.  To model this in Netbox, I add the 3 physical interfaces to the Device Type, and then I add a virtual interface on each Device called "IPMI Shared" with LAN 1 as the parent.  I would prefer modeling the virtual interface on the Device Type, but then there is no way to show its relation to LAN 1.

<img width="977" alt="image" src="https://github.com/user-attachments/assets/30a18a9b-0205-4571-ae90-615dbd9e139a">

**Comment by candlerb** [comment]
>  I would prefer modeling the virtual interface on the Device Type, but then there is no way to show its relation to LAN 1.

You mean, you'd like to be able to set a Parent interface in the Interface template in the Device Type? That seems like a reasonable request to me, especially since you can already link interface templates to a "Bridge":

![image](https://github.com/user-attachments/assets/314556fa-0e85-4be8-8900-4074dd5f6995)

Adding "Parent interface" and "LAG interface" here for consistency seems reasonable.

However, you are getting into slightly murky waters, because of Netbox's conflation of physical ports and logical interfaces. Physical ports are clearly part of the device type: nobody goes up to a device with a drill and adds a new port to a device. But logical interfaces, and their relationship to physical ports, is dynamic and configurable.

Hence arguably bridges, subinterfaces, loopbacks and LAGs are not part of the device type.  But pragmatically, you're saying "we get devices of type X and they are preconfigured by the manufacturer in this particular way" and I can't argue with that - just like switches that come preconfigured with a Vlan 1.

Another way you can deal with this is with a Custom Script which creates and customises the device interface in what ever way you like. But then you have to point people to use the Custom Script whenever creating one of these devices.

**Reply by ZPrimed** [reply]
I'm not the OP here, but this functionality would be very useful to me as well. 

The majority of devices I deploy are effectively "copies" of a template device, and part of that template is several (virtual) sub-interfaces attached to specific VLANs... but DeviceTypes don't currently allow for this. 

Obviously we can't spawn an instance with VLANs pre-configured, but it would still save time to be able to have the Virtual interfaces pre-defined and pre-linked to their parents, since they are going to be the same across 99% of the devices I deploy.
